src/chunkserver/copyset_node.cpp

```diff
@@ -322,9 +323,11 @@ void CopysetNode::on_apply(::braft::Iterator &iter) {
             auto chunkId = request.chunkid();
             concurrentapply_->Push(chunkId, ChunkOpRequest::Schedule(request.optype()),  // NOLINT
                                    &ChunkOpRequest::OnApplyFromLog, opReq,
-                                   dataStore_, std::move(request), data);
+                                   dataStore_, std::move(request), data, wrapperPtr);
         }
     }
+    // 等待写操作完成，否则on_apply结束后，异步有写错误无法调用set_error_and_rollback()
+    concurrentapply_->Flush();
```

Comment:
Will there be performance degradation here?

Reply:
I don't have any better ideas. When the on_apply() method completes, last_valid_index will be updated and the Iterator will be destroyed, but concurrent tasks may not be completed yet. Calling iterator->set_error_and_callback() may fail when a write error occurs.